universal-application-tool-0.0.1/conf/routes

```diff
@@ -18,3 +20,12 @@ GET     /admin/create               controllers.admin.CreateAppController.index
 # Map static resources from the /public folder to the /assets URL path
 GET     /assets/*file               controllers.Assets.versioned(file)
 
+# Authentication callback from federated auth server
+GET     /callback                    @org.pac4j.play.CallbackController.callback(request: Request)
+POST    /callback                    @org.pac4j.play.CallbackController.callback(request: Request)
+
+# Log into application
+GET     /loginForm                   controllers.HomeController.loginForm(request: Request)
+
+# Log out of application
+GET     /logout                      @org.pac4j.play.LogoutController.logout(request: Request)
```

Comment:
suggestion (blocking): since we have control of the routes, let's try to follow the routing convention. How about:

```
GET /federatedSession/new @org.pac4j.play.CallbackController.callback
POST    /federatedSession                    @org.pac4j.play.CallbackController.callback

GET /session/new  controllers.HomeController.loginForm
DELETE /session  @org.pac4j.play.LogoutController.logout
```

If the pac4j controller requires destroying the session to use a `GET` request that last one could instead be `GET /session/destroy @org.pac4j.play.LogoutController.logout`

Reply:
I don't think those are "actions on resources" the same way that the routing convention wants them to be - I see it in principle but it's pretty divergent from common practice, and I prefer common practice to consistency within the project.  Also, you seem to need to specify the same endpoint for both the POST and the GET on the callback controller, and /login and /logout are much clearer than /session/new and /session/destroy.  Plus, they're pages you'd want to be able to link to - you can't link to a `DELETE` request.

How strongly do you feel about it?

Reply:
> How strongly do you feel about it?

Not strongly, especially after finding out Play doesn't support pseudo-http methods via hidden form field (which is what some frameworks do to support `PUT` and `DELETE` without JS).

Feel free to ignore this comment.

Reply:
👍